sql/analyzer/aliases.go

```diff
@@ -116,15 +112,16 @@ func getTableAliases(n sql.Node, scope *Scope) (TableAliases, error) {
 			rt := getResolvedTable(node.Destination)
 			analysisErr = passAliases.add(rt, rt)
 			return false
-		case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode, *plan.RecursiveTable:
-			analysisErr = passAliases.add(node.(sql.Nameable), node.(sql.Nameable))
-			return false
 		case *plan.IndexedTableAccess:
 			rt := getResolvedTable(node.ResolvedTable)
 			analysisErr = passAliases.add(rt, node)
 			return false
+		case sql.Nameable:
+			analysisErr = passAliases.add(node, node)
+			return false
 		case *plan.UnresolvedTable:
 			panic("Table not resolved")
+		default:
```

Comment:
What's the reason for having an empty `default` case?

Reply:
Mostly a Go language standards/ hygiene thing I think, we explicitly want to not do anything on default. No guesswork for whether dev missed a case. We could fold the code after this into default, which the compiler probably does.